### PR TITLE
new entity + tests

### PR DESCRIPTION
**Pull request by Andrii-Trubetskoi**
@ahonymous @spolischook

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    - php: '7.1'
+      env: TARGET=phpunit
+
+before_script:
+  - composer self-update
+  - composer install
+
+script:
+  - if [ "$TARGET" = "phpunit" ]; then composer install; fi
+  - if [ "$TARGET" = "phpunit" ]; then phpunit -c phpunit.xml.dist --coverage-clover=coverage.clover; fi
```
Comment (ahonymous) [inline .travis.yml]: Better when you run tests after checking code styles

File: composer.json [hunk 2]
```diff
@@ -48,7 +53,7 @@
     },
     "config": {
         "platform": {
-            "php": "5.5.9"
+            "php": "7.1"
```
Comment (ahonymous) [inline composer.json:56]: You php version is 5.6. Or set needly version in the require section.

File: .travis.yml (hunk 1 above)
Comment (spolischook) [reply .travis.yml]: it's completely different checks. Travis-Ci give you possibility to run several builds for one check

**Comment by spolischook** [comment]
+12,642 −223  - it's hard to review huge amount of code. Please merge it and start a new one, with only one feature by PR

**Comment by spolischook** [comment]
merge it asap

**Comment by spolischook** [comment]
merge it asap

**Pull request merged**